client/components/ProductOverview/ThumbnailView.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState, useEffect } from 'react';
+import { AiOutlineArrowDown, AiOutlineArrowUp } from 'react-icons/ai';
+import PropTypes from 'prop-types';
+
+const ThumbnailView = ({ photos, handleClick, currentIndex }) => {
+  const [thumbnailStart, setThumbnailStart] = useState(currentIndex);
+  const [thumbnailEnd, setThumbnailEnd] = useState(7);
+
+  useEffect(() => {
+    if (photos.length <= 7) {
+      setThumbnailStart(0);
+      setThumbnailEnd(photos.length);
+    } else if (photos.length > 7) {
+      if (currentIndex > photos.length - 7) {
+        setThumbnailStart(photos.length - 7);
+        setThumbnailEnd(photos.length);
+      } else if (currentIndex < 7) {
+        setThumbnailStart(0);
+        setThumbnailEnd(7);
+      }
+    }
+  }, [currentIndex]);
+
+  const handleArrowUp = () => {
```

Comment:
Same with arrowLeft and arrowRight. Is there a way to have one handler that changes behavior based on a flag rather that additional functions?

Reply:
I think in this case the code is more readable with separate functions.